sources/packages/backend/libs/sims-db/src/entities/notification.model.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  OneToOne,
+  PrimaryGeneratedColumn,
+} from "typeorm";
+import { ColumnNames } from "../constant";
+import { Message } from "./message.model";
+import { RecordDataModel } from "./record.model";
+import { User } from "./user.model";
+
+@Entity({
+  name: "notifications",
+})
+export class Notification extends RecordDataModel {
+  /**
+   * Auto-generated sequential primary key column.
+   */
+  @PrimaryGeneratedColumn()
+  id: number;
+  /**
+   * User associated with this notification.
+   */
+  @OneToOne(() => User, { eager: false, cascade: false })
+  @JoinColumn({
+    name: "user_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  user: User;
+  /**
+   * Template id.
+   */
+  @Column({
+    name: "template_id",
+    nullable: false,
+  })
+  templateId: string;
+  /**
+   * GC notification payload.
+   */
+  @Column({
+    name: "gc_notify_payload",
+    type: "jsonb",
+    nullable: false,
+  })
+  gcNotifyPayload: unknown;
+  /**
+   * Message associated with this notification.
+   */
+  @OneToOne(() => Message, { eager: false, cascade: false })
+  @JoinColumn({
+    name: "message_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  message: Message;
+  /**
+   * Date Sent Column
+   */
+  @Column({
+    name: "date_sent",
+    type: "timestamptz",
+    nullable: false,
+  })
+  dateSent: Date;
+  /**
+   * Date Read Column
+   */
+  @Column({
+    name: "date_read",
+    type: "timestamptz",
+    nullable: true,
+  })
+  dateRead: Date;
+}
+
+/**
+ * Enumeration types for Messages.
+ */
+export enum MessageType {
+  /**
+   * Message type student file upload.
+   */
+  StudentFileUpload = 1,
```

Comment:
For the message type we are directly assigning the primary key value to an enum here.
@henryoforeh-aot  @andrewsignori-aot @guru-aot @ann-aot @andrepestana-aot are we ok with this approach as primary keys can be subjective to the existing data in table. I see that the primary key is INT and assigned by us but still.

Reply:
The code must know the message id in some way, so we will be using either an enum or some const. I am ok with the enum.

Reply:
me too and also its seeded

Reply:
I am also okay with this approach rather having a constant to fetch the message id.

Reply:
Going with team decision. Resolving.